[libexif] Update to 0.6.22

oskaritimperi opened the pull request:
**Describe the pull request**

- What does your PR fix?

Update libexif version from 0.6.21 to 0.6.22

From the [release page on github](https://github.com/libexif/libexif/releases/tag/libexif-0_6_22-release):

> It largely contains stability, security and bugfixes.
>
> Some EXIF 2.3 tags have been added.
>
> It is ABI compatible to 0.6.21 release.

- Which triplets are supported/not supported? Have you updated the CI baseline?

No change.

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?

Yes.

Commit:
[libexif] Update to 0.6.22

From the release page on github:

> It largely contains stability, security and bugfixes.
>
> Some EXIF 2.3 tags have been added.
>
> It is ABI compatible to 0.6.21 release.

c72578 reviewed (changes requested):
In **`portfile.cmake`**:
`include(vcpkg_common_functions)` is deprecated.
Could you please remove this line.

Please change the following line:
`file(INSTALL ${SOURCE_PATH}/COPYING DESTINATION ${CURRENT_PACKAGES_DIR}/share/libexif RENAME copyright)`
->
`file(INSTALL ${SOURCE_PATH}/COPYING DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)`

Commit:
[libexif] Fixes to portfile.cmake

- don't include vcpkg_commong_functions
- use the PORT variable when installing copyright file

NancyLi1013 commented on ports/libexif/portfile.cmake:
```diff
@@ -1,14 +1,12 @@
-include(vcpkg_common_functions)
-
 if(VCPKG_CMAKE_SYSTEM_NAME STREQUAL "WindowsStore")
     message(FATAL_ERROR "libexif currently only supports being built for desktop")
 endif()
```
```suggestion
vcpkg_fail_port_install(ON_TARGET "uwp")
```
We prefer to use this function to post the message. Could you please update this?

oswjk replied:
Done. What about the `Supports` field in the `CONTROL` file? Does that relate to this? I guess I shouldn't use that if you are phasing out the `CONTROL` files and moving to the manifests at some point?

NancyLi1013 replied:
This is related with `Supports `field in the `CONTROL` file. We might drop out `CONTROL` file in the near future but not for now.
So you don't need to change anything about this.

Commit:
[libexif] Use vcpkg_fail_port_install

NancyLi1013 commented:
@oswjk 
Is this PR ready to merge now?

@c72578 
Thanks for your help to review. It's so nice.

ras0219-msft commented:
LGTM, thanks for the update!
